mrbgems/mruby-io/src/io.c

```diff
@@ -530,6 +530,54 @@ mrb_io_s_popen(mrb_state *mrb, mrb_value klass)
 }
 #endif
 
+static int
+mrb_dup(mrb_state *mrb, int fd)
+{
+  int new_fd;
+
+  if (fd < 0)
+    return fd;
+
+  new_fd = dup(fd);
+  if (new_fd == -1)
+    mrb_sys_fail(mrb, 0);
+  return new_fd;
+}
+
+mrb_value
+mrb_io_initialize_copy(mrb_state *mrb, mrb_value copy)
+{
+  mrb_value orig;
+  mrb_value buf;
+  struct mrb_io *fptr_copy;
+  struct mrb_io *fptr_orig;
+
+  mrb_get_args(mrb, "o", &orig);
+  fptr_copy = (struct mrb_io *)DATA_PTR(copy);
+  if (fptr_copy != NULL) {
+    fptr_finalize(mrb, fptr_copy, FALSE);
+    mrb_free(mrb, fptr_copy);
+  }
+  fptr_copy = (struct mrb_io *)mrb_io_alloc(mrb);
```

Comment:
`fptr_copy` will not be freed (causes memory leakage) when an exception is raised on 562-576.